src/main/webapp/app/instructor-dashboard/instructor-course-dashboard.component.ts

```diff
@@ -43,80 +60,657 @@ export class InstructorCourseDashboardComponent implements OnInit, OnDestroy {
+                    'successfulTotal': 0,
+                    'successfullyCompletedTotalInPercent': 0,
+                    'participatedTotal': 0,
+                    'participationTotalInPercent': 0
+                };
+            }
+            finalScores[studentsQuizScore.id].QuizTotalScore = studentsQuizScore.totalScore;
+            finalScores[studentsQuizScore.id].QuizEveryScore = quizEveryScore;
+            finalScores[studentsQuizScore.id].QuizScoreString = quizScoreString;
+            finalScores[studentsQuizScore.id].participatedQuizzes = studentsQuizScore.participated;
+            finalScores[studentsQuizScore.id].successfulQuizzes = studentsQuizScore.successful;
+        }
+
+        for (const studentsModelingScore of this.typeModelingExercise) {
+
+            if (!finalScores[studentsModelingScore.id]) {
+                finalScores[studentsModelingScore.id] = {
+                    'firstName': studentsModelingScore.firstName,
+                    'lastName': studentsModelingScore.lastName,
+                    'login': studentsModelingScore.login,
+                    'email': studentsModelingScore.email,
+                    'QuizTotalScore': 0,
+                    'QuizEveryScore': {},
+                    'QuizScoreString': '',
+                    'ProgrammingTotalScore': 0,
+                    'ProgrammingEveryScore': {},
+                    'ProgrammingScoreString': '',
+                    'modelingTotalScore': 0,
+                    'modelingEveryScore': {},
+                    'modelingScoreString': '',
+                    'OverallScore': 0,
+                    'participatedQuizzes': 0,
+                    'participationQuizzesInPercent': 0,
+                    'successfulQuizzes': 0,
+                    'successfullyCompletedQuizzesInPercent': 0,
+                    'participatedProgramming': 0,
+                    'participationProgrammingInPercent': 0,
+                    'successfulProgramming': 0,
+                    'successfullyCompletedProgrammingInPercent': 0,
+                    'participatedModeling': 0,
+                    'participationModelingInPercent': 0,
+                    'successfulModeling': 0,
+                    'successfullyCompletedModelingInPercent': 0,
+                    'successfulTotal': 0,
+                    'successfullyCompletedTotalInPercent': 0,
+                    'participatedTotal': 0,
+                    'participationTotalInPercent': 0
+                };
+            }
+
+            let modelingScoreString = '';
+            let modelingEveryScore = {};
+
+            for (const modelings in this.allModelingExercises) {
+                let bool = true;
+                const exId = this.allModelingExercises[modelings].exId;
+
+                for (const scores in studentsModelingScore.scoreListM) {
+                    const exID = studentsModelingScore.scoreListM[scores].exID;
+                    if (exId === exID) {
```

Comment:
don't use exId and exID at the same time, this is very dangerous code, because you can easily have a mistake here